assignment3/src/pages/mutual-funds/search.tsx

```diff
@@ -0,0 +1,182 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { Link as RouterLink, useLocation } from "react-router-dom";
+import {
+  Box,
+  CircularProgress,
+  TextField,
+  Typography,
+  Grid,
+  Button,
+  Link,
+  List,
+  ListItem,
+  InputAdornment,
+  IconButton,
+} from "@mui/material";
+import { Search } from '@mui/icons-material';
+import { sleep } from "../../utils";
+
+type SearchResult = {
+  code: string;
+  name: string;
+};
+
+interface SearchResponse {
+  schemeCode: string;
+  schemeName: string;
+}
+
+export const Searchfunc = () => {
+  const [results, setResults] = useState<SearchResult[]>([]);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [emptyResult, setEmptyResult] = useState("No Results!");
+  const [searchWord, setSearchWord] = useState("");
+
+  const location = useLocation();
+
+  useEffect(() => {
+    setEmptyResult("Search for an Item!");
+  }, [])
+
+  const openFirstResult = async (searchWord: string) => {
+    console.log(searchWord);
+    let resultList = await getSearchResults(searchWord);
+    let firstResult = resultList?.at(0)?.code;
+    console.log(location.pathname);
+    if (firstResult) {
+      //console.log(firstResult);
+      window.location.href = location.pathname + "/" + firstResult;
```

Comment:
Don't use `window.location.href` - this causes a page reload and messes up the react history. Check `react-router-dom` API and use it to change path from within the react app.